code/go/0chain.net/smartcontract/dbs/event/eventdb.go

```diff
@@ -15,17 +20,81 @@ func NewEventDb(config dbs.DbAccess) (*EventDb, error) {
 		return nil, err
 	}
 	eventDb := &EventDb{
-		Store: db,
+		Store:          db,
+		eBufferChannel: make(chan eventCtx, 100),
+		eChannel:       make(chan eventCtx, 100),
 	}
 
 	if err := eventDb.AutoMigrate(); err != nil {
 		return nil, err
 	}
+	go eventDb.channelBufferIntermediate()
+	go eventDb.addEventWorker()
```

Comment:
this should be run with common.GetRootContext()

Reply:
done